src/erlcloud_aws.erl

```diff
@@ -664,6 +690,21 @@ get_metadata_credentials(Config) ->
             get_credentials_from_metadata(Config)
     end.
 
+-spec get_task_metadata_credentials(aws_config()) -> {ok, #metadata_credentials{}} | {error, term()}.
+get_task_metadata_credentials(Config) ->
+    %% See if we have cached credentials
+    case application:get_env(erlcloud, task_metadata_credentials) of
```

Comment:
not sure why we need this. it's either in ECS or in EC2 so reusing `metadata_credentials` should be enough